apps/src/blockly/customBlocks/googleBlockly/behaviorBlocks.js

```diff
@@ -0,0 +1,256 @@
+    text: msg.createBlocklyBehavior(),
+    callbackKey: 'createNewBehavior',
+  };
+
+  const behaviorDefinitionBlock = {
+    kind: 'block',
+    type: 'behavior_definition',
+    fields: {
+      NAME: Blockly.Msg.PROCEDURES_DEFNORETURN_PROCEDURE,
+    },
+  };
+
+  const createNewBehavior = function () {
+    // Everything here is place-holder code that should be replaced with a
+    // call to open the behavior editor with a new defintion block.
+    // Until then, we just create a block under all existing blocks on the
+    // main workspace.
+
+    createAndCenterNewDefBlock(behaviorDefinitionBlock);
+  };
+  if (useModalFunctionEditor) {
+    workspace.registerButtonCallback('createNewBehavior', createNewBehavior);
+    blockList.push(newBehaviorButton);
+  } else {
+    blockList.push(behaviorDefinitionBlock);
+  }
+  blockList.push(...getCustomCategoryBlocksForFlyout('Behavior'));
+
+  const allWorkspaces = Blockly.Workspace.getAll().filter(
+    workspace => !workspace.isFlyout
+  );
+  const allBehaviorNames = [];
+  allWorkspaces.forEach(workspace => {
+    const behaviorBlocks = workspace
+      .getTopBlocks()
+      .filter(topBlock => topBlock.type === 'behavior_definition');
+    behaviorBlocks.forEach(block =>
+      allBehaviorNames.push(block.getFieldValue('NAME'))
+    );
+  });
+  allBehaviorNames.sort().forEach(name => {
+    blockList.push({
+      kind: 'block',
+      type: 'gamelab_behavior_get',
+      extraState: {
+        name: name,
+      },
+      fields: {
+        NAME: name,
+      },
+    });
+  });
+
+  return blockList;
+}
+
+function getCustomCategoryBlocksForFlyout(category) {
+  const parser = new DOMParser();
+  const xmlDoc = parser.parseFromString(Blockly.toolboxBlocks, 'text/xml');
+
+  const categoryNodes = xmlDoc.getElementsByTagName('category');
+  for (const categoryNode of categoryNodes) {
+    const categoryCustom = categoryNode.getAttribute('custom');
+
+    if (categoryCustom === category) {
+      const xmlRootElement = xmlDoc.createElement('xml'); // Create a new <xml> root element
+
+      const blockNodes = categoryNode.getElementsByTagName('block');
+      for (const blockNode of blockNodes) {
+        if (blockNode.parentElement === categoryNode) {
+          xmlRootElement.appendChild(blockNode.cloneNode(true)); // Append cloned block nodes
+        }
+      }
+
+      const jsonBlocks = convertXmlToJson(xmlRootElement);
+
+      const flyoutBlocks = jsonBlocks.blocks.blocks.map(block =>
+        simplifyBlockStateForFlyout(block)
+      );
+      return flyoutBlocks; // Return the new <xml> root element with block children
+    }
+  }
+
+  return null; // Return null if the desired category is not found
+}
+
+// Used to simplify block state for inclusion in the Behaviors category flyout.
+function simplifyBlockStateForFlyout(block) {
+  // Clone the original block object to avoid modifying it directly
+  const modifiedBlock = {...block};
+
+  // Remove id, x, and y properties
```

Comment:
why do we remove id?

Reply:
From conversation with Maribeth: `It shouldn't matter, I think. When you create a new block from the flyout, a new copy of the block gets created on the workspace. the first copy of it you create will have the same id as the one in the flyout (if it has an id). for every subsequent block, there will be a block id collision so we'll just generate a new id for it. so i don't think it really matters either way.`